server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java

```diff
@@ -0,0 +1,150 @@
+package org.infinispan.server.core;
+
+import java.net.InetSocketAddress;
+import javax.management.DynamicMBean;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.factories.components.ManageableComponentMetadata;
+import org.infinispan.jmx.JmxUtil;
+import org.infinispan.jmx.ResourceDMBean;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.server.core.configuration.ProtocolServerConfiguration;
+import org.infinispan.server.core.logging.Log;
+import org.infinispan.server.core.transport.NettyTransport;
+
+/**
+ * A common protocol server dealing with common property parameter validation and assignment and transport lifecycle.
+ *
+ * @author Galder Zamarreño
+ * @author wburns
+ * @since 4.1
+ */
+public abstract class AbstractProtocolServer<A extends ProtocolServerConfiguration> extends AbstractCacheIgnoreAware
+        implements ProtocolServer<A> {
+
+   private final Log log = LogFactory.getLog(getClass(), Log.class);
+
+   private final String protocolName;
+
+   protected NettyTransport transport;
+   protected EmbeddedCacheManager cacheManager;
+   protected A configuration;
+   private ObjectName transportObjName;
+   private MBeanServer mbeanServer;
+
+   protected AbstractProtocolServer(String protocolName) {
+      this.protocolName = protocolName;
+   }
+
+   protected void startInternal(A configuration, EmbeddedCacheManager cacheManager) {
+      this.configuration = configuration;
+      this.cacheManager = cacheManager;
+
+      if (log.isDebugEnabled()) {
+         log.debugf("Starting server with configuration: %s", configuration);
```

Comment:
This is just how it was before, tbh it probably doesn't really matter since this is only ever invoked once.  The debug enabled is probably just a tiny fraction faster when it isn't enabled.